libs/langchain-google-cloud-sql-pg/src/loader.ts

```diff
@@ -0,0 +1,180 @@
+  tableName?: string;
+  schemaName?: string;
+  contentColumns?: string[];
+  metadataColumns?: string[];
+  format?: "text" | "json" | "yaml" | "csv";
+  formatter?: (row: Row, contentColumns: string[]) => string;
+  query?: string;
+  metadataJsonColumn?: string | null;
+}
+
+
+function parseDocFromRow(
+  contentColumns: string[],
+  metadataColumns: string[],
+  row: Row,
+  metadataJsonColumn: string | null = DEFAULT_METADATA_COL,
+  formatter: (row: Row, contentColumns: string[]) => string = textFormatter
+): Document {
+  const pageContent = formatter(row, contentColumns);
+  const metadata: { [key: string] } = {};
+
+  if (metadataJsonColumn && row[metadataJsonColumn]) {
+    Object.entries(row[metadataJsonColumn]).forEach(([k, v]) => {
+      metadata[k] = v;
+    });
+  }
+
+  metadataColumns.forEach(column => {
+    if (column in row && column !== metadataJsonColumn) {
+      metadata[column] = row[column];
+    }
+  });
+
+
+  return { pageContent, metadata };
+}
+
+export class PostgresLoader extends BaseDocumentLoader {
+
+  private engine: PostgresEngine;
+
+  tableName?: string;
+
+  schemaName?: string;
+
+  contentColumns?: string[];
+
+  metadataColumns?: string[];
+
+  format?: "text" | "json" | "yaml" | "csv";
+
+  formatter?: (row: Row, contentColumns: string[]) => string;
+
+  query?: string;
+
+  metadataJsonColumn?: string | null;
+
+  constructor(engine: PostgresEngine, options: PostgresLoaderOptions) {
+    super();
+    this.engine = engine;
+    this.contentColumns = options.contentColumns;
+    this.metadataColumns = options.metadataColumns;
+    this.query = options.query;
+    this.formatter = options.formatter;
+    this.metadataJsonColumn = options.metadataJsonColumn;
+  }
+
+  static async create(engine: PostgresEngine, { schemaName = 'public', tableName, contentColumns, metadataColumns, format, query, formatter, metadataJsonColumn }: PostgresLoaderOptions): Promise<PostgresLoader> {
+
+    if (tableName && query) {
+      throw new Error("Only one of 'table_name' or 'query' should be specified.");
+    }
+    if (!tableName && !query) {
+      throw new Error("At least one of the parameters 'table_name' or 'query' needs to be provided");
+    }
+    if (format && formatter) {
+      throw new Error("Only one of 'format' or 'formatter' should be specified.");
+    }
+
+    if (format && !["csv", "text", "json", "yaml"].includes(format)) {
+      throw new Error("format must be type: 'csv', 'text', 'json', 'yaml'");
+    }
+
+    let formatFunc = formatter;
+    if (formatFunc === undefined) {
+      if (format === "csv") {
+        formatFunc = csvFormatter;
+      } else if (format === "yaml") {
+        formatFunc = yamlFormatter;
+      } else if (format === "json") {
+        formatFunc = jsonFormatter;
+      } else {
+        formatFunc = textFormatter;
+      }
+    }
+
+    let queryStmt = query;
+    if (!queryStmt) {
+      queryStmt = `SELECT * FROM "${schemaName}"."${tableName}"`;
```

Comment:
Probably good to set some kind of limit here?

Reply:
The user can use the table argument to pull the whole table. I will add a note that we suggest using the lazyLoad.